File: src/main/java/net/reflact/engine/spells/dynamic/SpellEffect.java
package net.reflact.engine.spells.dynamic;

import com.google.gson.JsonObject;

public interface SpellEffect {
    /**
     * Executes the effect.
     * @param context The context of the cast.
     */
    void execute(CastContext context);

    /**
     * Loads configuration from JSON.
     */
    void load(JsonObject config);
}

File: src/main/java/net/reflact/engine/spells/dynamic/SpellEffect.kt
package net.reflact.engine.spells.dynamic

import com.google.gson.JsonObject

interface SpellEffect {
    /**
     * Executes the effect.
     * @param context The context of the cast.
     */
    fun execute(context: CastContext)

    /**
     * Loads configuration from JSON.
     */
    fun load(config: JsonObject)
}
